sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -55,6 +55,7 @@ statement
         rowFormat?  createFileFormat? locationSpec?
         (TBLPROPERTIES tablePropertyList)?
         (AS? query)?                                                   #createTable
+    | createTableHeader LIKE source=tableIdentifier                    #createTableLike
```

Comment:
(For that reason I don't think we should use `createTableHeader` here since it doesn't make sense to say `CREATE TEMPORARY TABLE x LIKE ...`, and we just throw an exception downstream anyway.